tests/secret-providers/aws_secretmanager/aws_secretmanager_test.go

```diff
@@ -0,0 +1,452 @@
+//go:build e2e
+// +build e2e
+
+package aws_secret_manager_test
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "aws-secret-manager-test"
+)
+
+var (
+	testNamespace              = fmt.Sprintf("%s-ns", testName)
+	deploymentName             = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName           = fmt.Sprintf("%s-so", testName)
+	triggerAuthenticationName  = fmt.Sprintf("%s-ta", testName)
+	secretName                 = fmt.Sprintf("%s-secret", testName)
+	postgreSQLStatefulSetName  = "postgresql"
+	postgresqlPodName          = fmt.Sprintf("%s-0", postgreSQLStatefulSetName)
+	postgreSQLUsername         = "test-user"
+	postgreSQLPassword         = "test-password"
+	postgreSQLDatabase         = "test_db"
+	postgreSQLConnectionString = fmt.Sprintf("postgresql://%s:%s@postgresql.%s.svc.cluster.local:5432/%s?sslmode=disable",
+		postgreSQLUsername, postgreSQLPassword, testNamespace, postgreSQLDatabase)
+	minReplicaCount = 0
+	maxReplicaCount = 2
+
+	awsRegion                = os.Getenv("TF_AWS_REGION")
+	awsAccessKeyID           = os.Getenv("TF_AWS_ACCESS_KEY")
+	awsSecretAccessKey       = os.Getenv("TF_AWS_SECRET_KEY")
+	awsCredentialsSecretName = fmt.Sprintf("%s-credentials-secret", testName)
```

Comment:
As AWS can't be created/removed in a short period, we need to use dynamic naming for it. Something like:
```suggestion
	awsCredentialsSecretName = fmt.Sprintf("%s-credentials-secret-%d", testName, GetRandomNumber())
```